### PR TITLE
Feature/testing infrastructure

### PR DESCRIPTION
**Pull request by mauriciovieira**
Following Rocket.Chat testing infrastructure. 

What I have now is an end-to-end test using chimp.js that opens a browser and tests the login page. The test is currently flaky, not returning positive answer all times.

I would also like to make it to run in Travis, but it'd be better as a new task.

### CODE REVIEW

**Comment by mauriciovieira** [comment]
End-to-end tests for login page working:

screen shot 2017-05-05 at 18 52 54

But it requires the application to be running on other terminal.

Rocket.chat solves it in travis-ci with [xfvb](https://en.wikipedia.org/wiki/Xvfb)

File: tests/chimp-config.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 */
+
+
+// import path from 'path';
+// import {isCI} from '../lib/ci';
+
+module.exports = {
+// 	// - - - - CHIMP - - - -
+// 	watch: false,
```
Comment (brylie) [inline tests/chimp-config.js]: Clean up unused code (comments)?

**Comment by mauriciovieira** [comment]
After trying for a few weeks, I coudn't reliably test the Branding page following the [official docs](https://guide.meteor.com/testing.html). 

What I am trying to do here is to create a basic set of end to end tests inspired by Rocket.Chat [tests](https://github.com/RocketChat/Rocket.Chat/tree/079b7ff7c44b2978a353eef828e9818661bef110/tests/end-to-end/ui). 

Then, I will test the Branding page to close #1506.

@ashakunt @brylie Notice that we'll still have to run the end2end tests on travis, probably also following the Rocket.Chat [example](https://travis-ci.org/RocketChat/Rocket.Chat/builds/231316389#L2105). 

These e2e tests can add some regression tests on basic behavior, but we'll need a solution for testing the REST API as well as other server side methods (covered by #1714).

**Comment by mauriciovieira** [comment]
@brylie @ashakunt  I added documentation on an initial suite of end to end tests. In order to run, it is necessary to have a meteor server running and also google chrome.

Please review. 

This PR closes #2615

**Comment by mauriciovieira** [comment]
@brylie I fixed your [requests](https://github.com/apinf/platform/pull/2522#pullrequestreview-36963026)

**Comment by mauriciovieira** [comment]
Closes #2622 as well

**Comment by ashakunt** [comment]
Looking into it during today.

**Comment by mauriciovieira** [comment]
@wuder4 please also take a look.

**Comment by ashakunt** [comment]
I reviewed this branch using my browser today (ideally, would have wanted to run stuff locally); and oh boy, wasn't it pain to review such a large chunk in Chrome. 

In principle, I think its Ok to merge. However, a few things I would like to bring to notice: The testing team is looking forward to write the E2E tests in Robot + Selenium because they are easy to learn, write and run. So at this point, most of our test assets will start to be shown in that framework (at least all what will be contributed by @wuder4, unless, he is willing to learn chimp and already start with that?

IMHO, we should merge this to `develop` and try to develop our test assets using either CHIMP or ROBOT, what ever is preferred by the test writer. My personal fav is ROBOT because of its simplicity to read and write. Comments welcome.

**Comment by mauriciovieira** [comment]
@brylie @wuder4 do you have comments?

**Pull request merged**